### ISSUE TITLE
wait() suggestions

### ISSUE THREAD

**Issue by lukehorvat** [segmentio/nightmare#119]
I think the `value` parameter should be removed from `.wait(fn, value, [delay])`, and `fn` should simply be a function that returns a boolean. It's more readable because you can see the condition that determines whether the wait is over.

In other words, changing this:

```javascript
nightmare
.wait(function() {
  var thing = document.querySelector('.thing');
  return thing ? thing.textContent.toLowerCase() : 'goodbye';
}, 'hello', 6000);
```

...to this:

```javascript
nightmare
.wait(function() {
  var thing = document.querySelector('.thing');
  return thing && thing.textContent.toLowerCase() === 'hello';
}, 6000);
```

Also, I cannot see any reason for `delay` to be an *optional* parameter. If you don't specify a `delay`, the wait will be over after the first timeout, making the return value of `fn` inconsequential...

Furthermore, it would be nice if you could specify an indefinite wait that polls the page every X milliseconds, but *doesn't* refresh it. For example, there may be a lengthy operation that is running on the page that a refresh would cancel/reset. Perhaps add a `refresh` option?

So in conclusion, `.wait(fn, value, [delay])` → `.wait(fn, delay, [refresh])`.

Happy to send a pull request for any of this you agree with, if you want me to.

**Comment by reinpk** [comment]
in v2 (see #200) we've gone with your original suggestion :) thanks so much! the new function signature is `.wait(fn)` and waits until the function returns true after being evaluated on the page. v2 should be out tomorrow :)
